dot/utils_test.go

```diff
@@ -0,0 +1,607 @@
+					TelemetryURLs:  nil,
+					RetainBlocks:   0,
+					Pruning:        "",
+				},
+				Log: LogConfig{
+					CoreLvl:           3,
+					DigestLvl:         3,
+					SyncLvl:           3,
+					NetworkLvl:        3,
+					RPCLvl:            3,
+					StateLvl:          3,
+					RuntimeLvl:        3,
+					BlockProducerLvl:  3,
+					FinalityGadgetLvl: 3,
+				},
+				Init: InitConfig{Genesis: "./chain/gssmr/genesis-spec.json"},
+				Core: CoreConfig{
+					Roles:            4,
+					BabeAuthority:    true,
+					BABELead:         false,
+					GrandpaAuthority: true,
+					WasmInterpreter:  "wasmer",
+					GrandpaInterval:  1000000000,
+				},
+				Network: NetworkConfig{
+					Port:              7001,
+					Bootnodes:         nil,
+					ProtocolID:        "",
+					NoBootstrap:       false,
+					NoMDNS:            false,
+					MinPeers:          1,
+					MaxPeers:          50,
+					PersistentPeers:   nil,
+					DiscoveryInterval: 10000000000,
+				},
+				RPC: RPCConfig{
+					Enabled:        false,
+					External:       false,
+					Unsafe:         false,
+					UnsafeExternal: false,
+					Port:           8545,
+					Host:           "localhost",
+					Modules: []string{"system", "author", "chain", "state", "rpc", "grandpa", "offchain",
+						"childstate", "syncstate", "payment"},
+					WSPort:           8546,
+					WS:               false,
+					WSExternal:       false,
+					WSUnsafe:         false,
+					WSUnsafeExternal: false,
+				},
+				System: types.SystemInfo{},
+				State:  StateConfig{},
+			},
+			want1: &os.File{},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, got1 := newTestConfigWithFile(tt.args.t)
+
+			assert.Equal(t, tt.want, got)
+			if tt.want1 != nil {
+				assert.NotNil(t, got1)
+			}
+		})
+	}
+}
+
+func TestNewTestGenesis(t *testing.T) {
+	type args struct {
+		t *testing.T
+	}
+	tests := []struct {
+		name string
+		args args
+		want *genesis.Genesis
+	}{
+		{
+			name: "working example",
+			args: args{t: t},
+			want: &genesis.Genesis{},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got := NewTestGenesis(tt.args.t)
+			if tt.want != nil {
+				assert.NotNil(t, got)
+			}
+		})
+	}
+}
+
+func TestNewTestGenesisFile(t *testing.T) {
+	type args struct {
+		t   *testing.T
+		cfg *Config
+	}
+	tests := []struct {
+		name string
+		args args
+		want *os.File
+	}{
+		{
+			name: "working example",
+			args: args{
+				t:   t,
+				cfg: &Config{},
+			},
+			want: &os.File{},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got := newTestGenesisFile(tt.args.t, tt.args.cfg)
+			if tt.want != nil {
+				assert.NotNil(t, got)
+			}
+		})
+	}
+}
+
+func TestRandomNodeName(t *testing.T) {
+	tests := []struct {
+		name string
+		want string
+	}{
+		{
+			name: "working example",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got := RandomNodeName()
+			assert.Greater(t, len(got), 3)
```

Comment:
maybe assert using `assert.Regexp(t, got, `yourregex`)?

Reply:
Added Regexp to test this.